### PR TITLE
[Console] allow multiline responses to console questions

### PR DESCRIPTION
**Pull request by ramsey**
Documentation for new multiline responses feature proposed in https://github.com/symfony/symfony/pull/37683

### CODE REVIEW

File: components/console/helpers/questionhelper.rst [hunk 1]
```diff
@@ -234,6 +234,32 @@ You can also specify if you want to not trim the answer by setting it directly w
         $name = $helper->ask($input, $output, $question);
     }
 
+Accept Multiline Answers
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, the question helper stops reading user input when it receives a newline
```
Comment (noniagriconomie) [inline components/console/helpers/questionhelper.rst:240]: 👍 but missing an `versionadded: 5.2` directive here i think

File: components/console/helpers/questionhelper.rst [hunk 2]
```diff
@@ -234,6 +234,32 @@ You can also specify if you want to not trim the answer by setting it directly w
...
+
+    use Symfony\Component\Console\Question\Question;
+
+    // ...
+    public function execute(InputInterface $input, OutputInterface $output)
+    {
+        // ...
+        $helper = $this->getHelper('question');
+
+        $question = new Question('How do you solve world peace?');
```
Comment (noniagriconomie) [inline components/console/helpers/questionhelper.rst:253]: very inspirational quote for the day :)

File: components/console/helpers/questionhelper.rst [hunk 3]
```diff
@@ -234,6 +234,32 @@ You can also specify if you want to not trim the answer by setting it directly w
...
+        $question->setMultiline(true);
+
+        $name = $helper->ask($input, $output, $question);
```
Comment (noniagriconomie) [inline components/console/helpers/questionhelper.rst:256]: i would call the var $text as a text is generaly a multitile, $name more to be a uniq word :)

Comment (ramsey) [reply components/console/helpers/questionhelper.rst:256]: Oops! Copy-paste error. Thanks!

File: components/console/helpers/questionhelper.rst (hunk 2 above)
Comment (ramsey) [reply components/console/helpers/questionhelper.rst:253]: I figured it was appropriate for an answer that needs more than one line of input. 😉

File: components/console/helpers/questionhelper.rst (hunk 1 above)
Comment (ramsey) [reply components/console/helpers/questionhelper.rst:245]: Added in my latest push.

**Comment by ramsey** [comment]
Updated for changes made in https://github.com/symfony/symfony/pull/37683

**Review by fabpot — APPROVED** [review]

**Review by noniagriconomie — APPROVED** [review]

File: components/console/helpers/questionhelper.rst [hunk 4]
```diff
@@ -234,6 +234,36 @@ You can also specify if you want to not trim the answer by setting it directly w
...
+.. versionadded:: 5.2
+
+    The ``setMultiline()`` and ``isMultiline()`` methods were introduced in
+    Symfony 5.2.
+
+By default, the question helper stops reading user input when it receives a newline
+character (i.e., when the user hits ``ENTER`` once). However, you may specify that
+the response to a question should allow multiline answers by passing ``true`` to
+:method:`Symfony\\Component\\Console\\Question\\Question::setMultiline`::
+
```
Comment (noniagriconomie) [inline components/console/helpers/questionhelper.rst:249]: i would just add somethin like

```php
// src/Command/MyCommand.php
```

File: components/console/helpers/questionhelper.rst [hunk 5]
```diff
@@ -234,6 +234,36 @@ You can also specify if you want to not trim the answer by setting it directly w
...
+    public function execute(InputInterface $input, OutputInterface $output)
+    {
+        // ...
+        $helper = $this->getHelper('question');
+
+        $question = new Question('How do you solve world peace?');
+        $question->setMultiline(true);
+
+        $answer = $helper->ask($input, $output, $question);
+    }
```
Comment (noniagriconomie) [inline components/console/helpers/questionhelper.rst:262]: missing a return here if i am not mistaken

File: components/console/helpers/questionhelper.rst (hunk 4 above)
Comment (ramsey) [reply components/console/helpers/questionhelper.rst:249]: I'm not sure what you mean.

File: components/console/helpers/questionhelper.rst (hunk 5 above)
Comment (ramsey) [reply components/console/helpers/questionhelper.rst:262]: I copied from other examples in the docs, none of which include a return or `// ...` at the end of the method.

File: components/console/helpers/questionhelper.rst (hunk 4 above)
Comment (ramsey) [reply components/console/helpers/questionhelper.rst:249]: Never mind. I see what you mean now. The context for your comment didn't show the lines below, so I thought you were talking about the lines above. 😁 

None of the other examples on this page show an example name for their command script. If I add this here, should I also add it to the rest of the examples?

Comment (noniagriconomie) [reply components/console/helpers/questionhelper.rst:249]: ho i did not figure that other have not
but yes i see a lot on code example the path file on top as a comment
it helps understanding in which file the class is and its meaning etc :) 

but i do not know if you should update all others :s

Comment (ramsey) [reply components/console/helpers/questionhelper.rst:249]: I don't disagree, but it's inconsistent with the rest of the examples in this file.

**Review by OskarStark — APPROVED** [review]

**Review by maxhelias — APPROVED** [review]

**Pull request merged**